engine/src/main/java/org/camunda/bpm/engine/impl/cmd/FetchExternalTasksCmd.java

```diff
@@ -74,12 +76,24 @@ public FetchExternalTasksCmd(String workerId, int maxResults, Map<String, TopicF
     for (ExternalTaskEntity entity : externalTasks) {
       
       TopicFetchInstruction fetchInstruction = fetchInstructions.get(entity.getTopicName());
-      entity.lock(workerId, fetchInstruction.getLockDuration());
-
-      LockedExternalTaskImpl resultTask = LockedExternalTaskImpl.fromEntity(entity, fetchInstruction.getVariablesToFetch(), fetchInstruction.isLocalVariables(),
-          fetchInstruction.isDeserializeVariables(), fetchInstruction.isIncludeExtensionProperties());
-
-      result.add(resultTask);
+      //retrieve latest execution to avoid concurrent modifications to the execution @https://jira.camunda.com/browse/CAM-10750
+      ExecutionEntity execution = commandContext
+    		  .getExecutionManager()
+    		  .findExecutionById(entity.getExecutionId());
+      if(null!=execution){
+    	  entity.setExecution(execution);
+    	  entity.lock(workerId, fetchInstruction.getLockDuration());
+      }
+      
+      try {
+          LockedExternalTaskImpl resultTask = LockedExternalTaskImpl.fromEntity(entity, fetchInstruction.getVariablesToFetch(), fetchInstruction.isLocalVariables(),
+                  fetchInstruction.isDeserializeVariables(), fetchInstruction.isIncludeExtensionProperties());
+
+          result.add(resultTask); 
+      } catch(NullValueException e) {
+    	  //catch concurrent workers trying to fetchandLock scenario by handling exception @https://jira.camunda.com/browse/CAM-10750
+    	  LOG.logTaskAlreadyFetched(workerId, e);
```

Comment:
I think the try-catch is not. Just move the conversion to `LockedExternalTaskImpl` into the `if` block above. The engine caches an execution once retrieved, so it will not make a second SELECT that could return `null`.

Reply:
In the scenario we tested, we had 8 worker nodes running in parallel. Some were still throwing NullValueException hence had to add this catch block. Will move the LockedExternalTaskImpl.fromEntity inside the if and retain NullValueException. Let me know if this sounds good to you